src/libbson/src/bson/bson.c

```diff
@@ -2140,15 +2179,24 @@ bson_reserve_buffer (bson_t *bson, uint32_t size)
       return NULL;
    }
 
-   if (!_bson_grow (bson, size)) {
+   /* The caller wants a total document size of "size".
```

Comment:
Consider renaming the `size` parameters in `_bson_grow` and `bson_reserver_buffer` (e.g. to `grow_size` and `total_size`?) to help clarify the distinction.